packages/contracts/contracts/L1/ValidatorPool.sol

```diff
@@ -198,14 +248,51 @@ contract ValidatorPool is ReentrancyGuardUpgradeable, Semver {
      */
     function _tryUnbond() private returns (bool) {
         uint256 outputIndex = nextUnbondOutputIndex;
-        address recipient = L2_ORACLE.getSubmitter(outputIndex);
 
         Types.Bond storage bond = bonds[outputIndex];
-        if (block.timestamp >= bond.expiresAt && bond.amount > 0) {
-            _increaseBalance(recipient, bond.amount);
-            nextUnbondOutputIndex = nextUnbondOutputIndex + 1;
-            emit Unbonded(outputIndex, recipient, bond.amount);
-            bond.amount = 0;
+        uint128 bondAmount = bond.amount;
+        if (block.timestamp >= bond.expiresAt && bondAmount > 0) {
+            unchecked {
+                ++nextUnbondOutputIndex;
+            }
+
+            delete bonds[outputIndex];
+
+            Types.CheckpointOutput memory output = L2_ORACLE.getL2Output(outputIndex);
+            _increaseBalance(output.submitter, bondAmount);
+            emit Unbonded(outputIndex, output.submitter, bondAmount);
+
+            // Select the next priority validator.
+            // TODO(pangssu): improve next validator selection
+            uint256 validatorIndex = uint256(
+                keccak256(abi.encodePacked(output.outputRoot, block.number, block.coinbase))
+            ) % validators.length;
+            nextPriorityValidator = validators[validatorIndex];
+
+            // Calculate the elapsed time to submitting the output.
+            uint256 penalty = 0;
+            uint256 elapsed = output.timestamp -
+                L2_ORACLE.computeL2Timestamp(output.l2BlockNumber + 1);
+            if (elapsed > ROUND_DURATION) {
+                elapsed -= ROUND_DURATION;
+            }
```

Comment:
Is it intended to decrease the value when elapsed time exceeds a certain threshold? Shouldn't it be set to a maximum value instead?

```suggestion
            if (elapsed > ROUND_DURATION) {
                elapsed = ROUND_DURATION;
            }
```

Reply:
Since both rounds have the same penalty duration, I subtracted ROUND_DURATION to simplify the calculation.